cli/functions/runHermesEmitBinaryCommand.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * code based on appcenter-cli
+ */
+
+const childProcess = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const shell = require('shelljs');
+
+/**
+ * Run Hermes compile CLI command
+ *
+ * @param bundleName {string} JS bundle file name
+ * @param outputPath {string} Path to output .hbc file
+ * @param sourcemapOutput {string} Path to output sourcemap file (Warning: if sourcemapOutput points to the outputPath, the sourcemap will be included in the CodePush bundle and increase the deployment size)
+ * @param extraHermesFlags {string[]} Additional options to pass to `hermesc` command
+ * @return {Promise<void>}
+ */
+async function runHermesEmitBinaryCommand(
+    bundleName,
+    outputPath,
+    sourcemapOutput,
+    extraHermesFlags = [],
+) {
+    /**
+     * @type {string[]}
+     */
+    const hermesArgs = [
+        '-emit-binary',
+        '-out',
+        path.join(outputPath, bundleName + '.hbc'),
+        path.join(outputPath, bundleName),
+        ...extraHermesFlags,
+    ];
+    if (sourcemapOutput) {
+        hermesArgs.push('-output-source-map');
+    }
+
+    console.log('Converting JS bundle to byte code via Hermes, running command:\n');
+
+    return new Promise((resolve, reject) => {
+        try {
+            const hermesCommand = getHermesCommand();
+
+            const disableAllWarningsArg = '-w';
+            shell.exec(`${hermesCommand} ${hermesArgs.join(' ')} ${disableAllWarningsArg}`);
+
+            // Copy HBC bundle to overwrite JS bundle
+            const source = path.join(outputPath, bundleName + '.hbc');
+            const destination = path.join(outputPath, bundleName);
+            shell.cp(source, destination);
+            shell.rm(source);
+            resolve();
+        } catch (e) {
+            reject(e);
+        }
+    }).then(() => {
+        if (!sourcemapOutput) {
+            // skip source map compose if source map is not enabled
+            return;
```

Comment:
I think using `async - await` is better than using `.then` in this case

Reply:
@changwoolab 😃 
This code follows the original code of `appcenter-cli`.
It feels somewhat awkward to read. But, I prefer to leave this code as it is, given its stability, and focus resources on moving forward.